pkg/move/runner.go

```diff
@@ -460,8 +461,53 @@ func (r *Runner) startBackgroundRoutines(ctx context.Context) {
 	go r.replClient.StartPeriodicFlush(ctx, repl.DefaultFlushInterval)
 	//go r.tableChangeNotification(ctx)
 
-	//go r.dumpStatus(ctx)                 // start periodically writing status
-	go r.dumpCheckpointContinuously(ctx) // start periodically dumping the checkpoint.
+	// Start go routines for checkpointing and dumping status
+	status.WatchTask(ctx, r, r.logger)
+}
+
+func (r *Runner) Status() string {
+	state := r.status.Get()
+	if state > status.CutOver {
+		return ""
+	}
+	switch state { //nolint: exhaustive
+	case status.CopyRows:
+		// Status for copy rows
+		return fmt.Sprintf("migration status: state=%s copy-progress=%s binlog-deltas=%v total-time=%s copier-time=%s copier-remaining-time=%v copier-is-throttled=%v",
+			r.status.Get().String(),
+			r.copier.GetProgress(),
+			r.replClient.GetDeltaLen(),
+			time.Since(r.startTime).Round(time.Second),
+			time.Since(r.copier.StartTime()).Round(time.Second),
+			r.copier.GetETA(),
+			r.copier.GetThrottler().IsThrottled(),
+		)
+	case status.WaitingOnSentinelTable:
+		return fmt.Sprintf("migration status: state=%s total-time=%s sentinel-wait-time=%s sentinel-max-wait-time=%s",
+			r.status.Get().String(),
+			time.Since(r.startTime).Round(time.Second),
+			time.Since(r.sentinelWaitStartTime).Round(time.Second),
+			sentinelWaitLimit,
+		)
+	case status.ApplyChangeset, status.PostChecksum:
+		// We've finished copying rows, and we are now trying to reduce the number of binlog deltas before
+		// proceeding to the checksum and then the final cutover.
+		return fmt.Sprintf("migration status: state=%s binlog-deltas=%v total-time=%s",
+			r.status.Get().String(),
+			r.replClient.GetDeltaLen(),
+			time.Since(r.startTime).Round(time.Second),
+		)
+	case status.Checksum:
+		// This could take a while if it's a large table.
+		return fmt.Sprintf("migration status: state=%s checksum-progress=%s binlog-deltas=%v total-time=%s checksum-time=%s",
+			r.status.Get().String(),
+			r.checker.GetProgress(),
+			r.replClient.GetDeltaLen(),
+			time.Since(r.startTime).Round(time.Second),
+			time.Since(r.checker.StartTime()).Round(time.Second),
+		)
+	}
+	return ""
```

Comment:
If you put the return statement inside a `default:` section of the case I think you wouldn't need to set //nolint:exhaustive.